src/lib.rs

```diff
@@ -42,10 +43,37 @@ pub enum KVMiInterceptType {
     Breakpoint = KVMI_EVENT_BREAKPOINT as isize,
     Pagefault = KVMI_EVENT_PF as isize,
 }
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u8)]
 pub enum KVMiPageAccess {
-    PageAccessW = kvmi_sys::KVMI_PAGE_ACCESS_W as isize,
-    PageAccessR = kvmi_sys::KVMI_PAGE_ACCESS_R as isize,
-    PageAccessX = kvmi_sys::KVMI_PAGE_ACCESS_X as isize,
+    NIL = 0,
+    R = KVMI_PAGE_ACCESS_R as u8,
+    W = KVMI_PAGE_ACCESS_W as u8,
+    RW = KVMI_PAGE_ACCESS_R as u8 | KVMI_PAGE_ACCESS_W as u8,
+    X = KVMI_PAGE_ACCESS_X as u8,
+    RX = KVMI_PAGE_ACCESS_R as u8 | KVMI_PAGE_ACCESS_X as u8,
+    WX = KVMI_PAGE_ACCESS_W as u8 | KVMI_PAGE_ACCESS_X as u8,
+    RWX = KVMI_PAGE_ACCESS_R as u8 | KVMI_PAGE_ACCESS_W as u8 | KVMI_PAGE_ACCESS_X as u8,
+}
+
+impl TryFrom<u32> for KVMiPageAccess {
+    type Error = &'static str;
+    fn try_from(access: u32) -> Result<Self, Self::Error> {
+        match access {
```

Comment:
What I meant was that you cast access here like this
```suggestion
        match access as u32 {
```

Reply:
The conversion should still be from `u8` to `KVMiPageAccess`